src/libraries/System.Private.Xml/src/System/Xml/Schema/XsdDateTime.cs

```diff
@@ -494,76 +494,79 @@ InternalKind switch
         /// </summary>
         public override string ToString()
         {
-            StringBuilder sb = new StringBuilder(64);
-            char[] text;
+            var vsb = new ValueStringBuilder(stackalloc char[64]);
             switch (InternalTypeCode)
             {
                 case DateTimeTypeCode.DateTime:
-                    PrintDate(sb);
-                    sb.Append('T');
-                    PrintTime(sb);
+                    PrintDate(ref vsb);
+                    vsb.Append('T');
+                    PrintTime(ref vsb);
                     break;
                 case DateTimeTypeCode.Time:
-                    PrintTime(sb);
+                    PrintTime(ref vsb);
                     break;
                 case DateTimeTypeCode.Date:
-                    PrintDate(sb);
+                    PrintDate(ref vsb);
                     break;
                 case DateTimeTypeCode.GYearMonth:
-                    text = new char[s_lzyyyy_MM];
-                    IntToCharArray(text, 0, Year, 4);
-                    text[s_lzyyyy] = '-';
-                    ShortToCharArray(text, s_lzyyyy_, Month);
-                    sb.Append(text);
-                    break;
+                    {
+                        Span<char> text = vsb.AppendSpan(s_lzyyyy_MM);
+                        IntToCharArray(ref text, 0, Year, 4);
+                        text[s_lzyyyy] = '-';
+                        ShortToCharArray(ref text, s_lzyyyy_, Month);
+                        break;
```

Comment:
Can this instead just be:
```C#
vsb.AppendSpanFormattable(Year);
vsb.Append('-');
vsb.AppendSpanFormattable(Month);
```
?
Similar question for the other such blocks below.

Reply:
done